### ISSUE TITLE
spring-cloud-aws-autoconfigure dependency

### ISSUE THREAD

**Issue by aminmc** [VanRoy/spring-data-jest#6]
Hi @VanRoy 

I was looking at making my app run in Docker. I noticed that AWSCredentialsProvider was being configured by Spring in org.springframework.cloud.aws.core.credentials.CredentialsProviderFactoryBean which creates AWSCredentialsProviderChain and my app fails to start up with:

```java
  throw new AmazonClientException("Unable to load AWS credentials from any provider in the chain");
```
Removing this:
```xml
<dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-aws-autoconfigure</artifactId>
            <version>1.1.0.RELEASE</version>
        </dependency>
```
leads to another issue.

I then changed added to my bean configuration:

```java
        @Bean
        @Primary
        public AWSCredentialsProvider awsCredentialsProvider() {
            return new DefaultAWSCredentialsProviderChain(); //This uses the environment variables for access key etc
        } 
```

And this fixed the problem.   What's the correct approach? 

Cheers

**Comment by VanRoy** [comment]
Hi @aminmc ,

spring-data-jest don't require spring-cloud-aws.

So the configuration depends of your requirement.

If your application already use or if require spring-cloud-aws, just configure your credentials based on information written in the docs ( http://cloud.spring.io/spring-cloud-static/spring-cloud-aws/1.0.4.RELEASE/#_spring_boot_auto_configuration ).

If your application not need spring-cloud-aws, do not import spring-cloud-aws dependency.
The bean DefaultAWSCredentialsProviderChain will be automatically created by spring-data-jest autoconfiguration ( https://github.com/VanRoy/spring-data-jest/blob/master/spring-boot-starter-data-jest/src/main/java/com/github/vanroy/springboot/autoconfigure/data/jest/ElasticsearchJestAWSAutoConfiguration.java#L47 ).

If you have some trouble with the configuration, please send me a sample project.

Julien.

**Comment by aminmc** [comment]
Hi @VanRoy thanks for getting back to me. The problem is that if I remove spring-cloud-aws then I get the following exception:
```java
INFO  o.s.core.annotation.AnnotationUtils - Failed to introspect annotations on [class com.github.vanroy.springboot.autoconfigure.data.jest.ElasticsearchJestAWSAutoConfiguration$$EnhancerBySpringCGLIB$$311083f2]: java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy

ERROR o.s.boot.SpringApplication - Application startup failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'requestMappingHandlerMapping' defined in class path resource [org/springframework/boot/autoconfigure/web/WebMvcAutoConfiguration$EnableWebMvcConfiguration.class]: Invocation of init method failed; nested exception is java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1578)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:545)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482)
    at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
    at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
    at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
    at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
    at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:776)
    at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:861)
    at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:541)
    at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122)
    at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:759)
    at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:369)
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:313)
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:1185)
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:1174)
    at io.arctech.solutions.squinky.api.SquinkyAPIServiceBootstrap.main(SquinkyAPIServiceBootstrap.java:15)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
Caused by: java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy
    at sun.reflect.annotation.AnnotationParser.parseClassArray(AnnotationParser.java:724)
    at sun.reflect.annotation.AnnotationParser.parseArray(AnnotationParser.java:531)
    at sun.reflect.annotation.AnnotationParser.parseMemberValue(AnnotationParser.java:355)
    at sun.reflect.annotation.AnnotationParser.parseAnnotation2(AnnotationParser.java:286)
    at sun.reflect.annotation.AnnotationParser.parseAnnotations2(AnnotationParser.java:120)
    at sun.reflect.annotation.AnnotationParser.parseAnnotations(AnnotationParser.java:72)
    at java.lang.Class.createAnnotationData(Class.java:3508)
    at java.lang.Class.annotationData(Class.java:3497)
    at java.lang.Class.createAnnotationData(Class.java:3513)
    at java.lang.Class.annotationData(Class.java:3497)
    at java.lang.Class.getAnnotation(Class.java:3402)
    at java.lang.reflect.AnnotatedElement.isAnnotationPresent(AnnotatedElement.java:258)
    at java.lang.Class.isAnnotationPresent(Class.java:3412)
    at org.springframework.core.annotation.AnnotatedElementUtils.hasAnnotation(AnnotatedElementUtils.java:617)
    at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping.isHandler(RequestMappingHandlerMapping.java:174)
    at org.springframework.web.servlet.handler.AbstractHandlerMethodMapping.initHandlerMethods(AbstractHandlerMethodMapping.java:213)
    at org.springframework.web.servlet.handler.AbstractHandlerMethodMapping.afterPropertiesSet(AbstractHandlerMethodMapping.java:184)
    at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping.afterPropertiesSet(RequestMappingHandlerMapping.java:128)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1637)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1574)
    ... 21 common frames omitted
```
And it seems to be related to ElasticsearchJestAWSAutoConfiguration 
```java
@AutoConfigureAfter(ContextRegionProviderAutoConfiguration.class)
```

I'll see if i can knock up a example project.

Cheers

**Comment by VanRoy** [comment]
Thanks @aminmc , I will be investigate on it.

**Comment by VanRoy** [comment]
@aminmc I did not succeeded to reproduce your error. Can you send me a sample project ?

**Comment by aminmc** [comment]
Sorry @VanRoy I'll do one as soon as possible.  Sorry for the delay.

**Comment by aminmc** [comment]
Hi @VanRoy sorry for not getting back to you sooner. Here is a simple project with spring-cloud-aws-autoconfigure removed. On app start up you should see the exception (DemoApplication).
[spring-cloud-aws-autoconfigure dependency.zip](https://github.com/VanRoy/spring-data-jest/files/466148/spring-cloud-aws-autoconfigure.dependency.zip)

**Comment by VanRoy** [comment]
@aminmc I think it's duplicated to #8.
It's fixed in 2.1.1-SNAPSHOT, please could you try it and tell me if the problem it's fixed.

Thank for your help.
Julien.

**Comment by aminmc** [comment]
@VanRoy Sorry again for taking this long. But I've confirmed issue resolved using 2.1.1-SNAPSHOT
